### PR TITLE
LOOP-1717: Fix settings navigation to not dismiss first

### PR DESCRIPTION
**Pull request by rickpasetto**
- Get rid of SettingsTableViewController (yay!)
- Use `show` instead of `present` to avoid the intermediate dismiss

(Note: because the contract for vended ViewControllers from plugins "implicitly" creates a UINavigationController, we can't use Swift UI navigation without major changes.)

[LOOP-1717](https://tidepool.atlassian.net/browse/LOOP-1717)

### CODE REVIEW

**Review by nhamming — APPROVED** [review]
Just wondering if we are deleting `SettingsTableViewControler`, should all unused Views/VCS based on the settings swap be removed (e.g., `SettingsImageTableViewCell`, `ExplicitlyDismissibleModal`, etc.)?

Otherwise LGTM.

File: Loop/Views/SettingsView.swift [hunk 1]
```diff
@@ -196,8 +189,6 @@ extension SettingsView {
     private var serviceChoices: [ActionSheet.Button] {
         var result = viewModel.servicesViewModel.inactiveServices.map { availableService in
             ActionSheet.Button.default(Text(availableService.localizedTitle)) {
-                // TODO: this "dismiss then call didTapAddService()" here is temporary, until we've completely gotten rid of SettingsTableViewController
-                self.dismiss()
                 self.viewModel.servicesViewModel.didTapAddService(availableService)
```
Comment (novalegra) [inline Loop/Views/SettingsView.swift:202]: When I tap to add the service and successfully add one, the view doesn't refresh (I have to exit the settings screen and then re-open it for the changes to show up)

**Review by novalegra — CHANGES_REQUESTED** [review]

File: Loop/Views/SettingsView.swift [hunk 2]
```diff
@@ -89,19 +90,22 @@ extension SettingsView {
...
-            return NavigationLink(destination: TherapySettingsView(
-                viewModel: TherapySettingsViewModel(mode: .settings,
-                                                    therapySettings: viewModel.therapySettings,
-                                                    supportedInsulinModelSettings: viewModel.supportedInsulinModelSettings,
-                                                    pumpSupportedIncrements: viewModel.pumpSupportedIncrements,
-                                                    syncPumpSchedule: viewModel.syncPumpSchedule,
-                                                    chartColors: .primary,
-                                                    didSave: viewModel.didSave))) {
-                LargeButton(action: { },
+            LargeButton(action: { self.showTherapySettings = true },
```
Comment (novalegra) [inline Loop/Views/SettingsView.swift:93]: Want to check it's intentional that this screen is now presented modally

File: Loop/Views/SettingsView.swift [hunk 3]
```diff
@@ -89,19 +90,22 @@ extension SettingsView {
...
                             includeArrow: false,
                             imageView: AnyView(Image("Therapy Icon")),
                             label: NSLocalizedString("Therapy Settings", comment: "Title text for button to Therapy Settings"),
                             descriptiveText: NSLocalizedString("Diabetes Treatment", comment: "Descriptive text for Therapy Settings"))
+                .sheet(isPresented: $showTherapySettings) {
+                    TherapySettingsView(
```
Comment (novalegra) [inline Loop/Views/SettingsView.swift:99]: I'm not seeing a title now for therapy settings (though it shows up fine for the pump simulator and Dexcom screens)

**Review by novalegra — COMMENTED** [review]
Noticed a few changes that I want to check were intentional :)

File: Loop/Views/SettingsView.swift (hunk 2 above)
Comment (rickpasetto) [reply Loop/Views/SettingsView.swift:93]: Yes.  This is very intentional ... based on Design feedback: they do not want some screens to show as a "left-right slide" navigation (TherapySettings) and others as a sheet (modal).  This makes them consistent.

File: Loop/Views/SettingsView.swift (hunk 3 above)
Comment (rickpasetto) [reply Loop/Views/SettingsView.swift:99]: Really?

Simulator Screen Shot - iPhone SE (2nd generation) - 2020-08-28 at 12 54 06

Shows fine for me...what am I missing?

Comment (novalegra) [reply Loop/Views/SettingsView.swift:99]: Hmm, showing up fine now when I re-checked-out. Probably had an incorrect branch checked out (I'm betting LoopKit)

Comment (rickpasetto) [reply Loop/Views/SettingsView.swift:99]: Yes `TherapySettingsView` is in `LoopKit[UI]`, so that would explain it 👍

**Review by novalegra — APPROVED** [review]
LGTM!

**Pull request merged**